packages/dev/core/src/Meshes/meshUtils.ts

```diff
@@ -0,0 +1,209 @@
+import type { AnimationGroup } from "../Animations/animationGroup";
+import { VertexBuffer } from "../Buffers/buffer";
+import { TmpVectors, Vector3 } from "../Maths/math.vector";
+import type { FloatArray, Nullable } from "../types";
+import type { AbstractMesh } from "./abstractMesh";
+
+function getExtentCorners(extent: { minimum: Vector3; maximum: Vector3 }): Array<Vector3> {
+    const minX = extent.minimum.x;
+    const minY = extent.minimum.y;
+    const minZ = extent.minimum.z;
+    const maxX = extent.maximum.x;
+    const maxY = extent.maximum.y;
+    const maxZ = extent.maximum.z;
+    return [
+        new Vector3(minX, minY, minZ),
+        new Vector3(maxX, maxY, maxZ),
+        new Vector3(maxX, minY, minZ),
+        new Vector3(minX, maxY, minZ),
+        new Vector3(minX, minY, maxZ),
+        new Vector3(maxX, maxY, minZ),
+        new Vector3(minX, maxY, maxZ),
+        new Vector3(maxX, minY, maxZ),
+    ];
+}
+
+/**
+ * Computes the maximum extents of the given meshes considering animation, skeleton, and morph targets.
+ * @param meshes The array of meshes to compute
+ * @param animationGroup An optional animation group to animate (must be started to take effect)
+ * @param animationStep An optional value indicating the number of seconds to step while looping through the given animation group
+ * @returns An array of world space extents corresponding to the given meshes
+ */
+export function computeMaxExtents(
+    meshes: Array<AbstractMesh>,
+    animationGroup: Nullable<AnimationGroup> = null,
+    animationStep = 1 / 6
+): Array<{ minimum: Vector3; maximum: Vector3 }> {
+    // Local vector to avoid allocations.
+    const position = TmpVectors.Vector3[0];
+
+    const meshExtents = new Map<number, { minimum: Vector3; maximum: Vector3 }>();
+    const skinnedMeshExtents = new Map<number, Map<number, { minimum: Vector3; maximum: Vector3 }>>();
+
+    // Compute the non-skinned and skinned mesh extents.
+    for (const mesh of meshes) {
+        const positions = mesh.getVerticesData(VertexBuffer.PositionKind);
+        if (!positions) {
+            continue;
+        }
+
+        // Initialize min/max positions with the original positions.
+        const numVertices = mesh.getTotalVertices();
+        const minPositions = new Array<Vector3>(numVertices);
+        const maxPositions = new Array<Vector3>(numVertices);

```

Comment:
Nit - those could be reused to avoid too much GC (and re-set here to `length = 0`)